CraftTweaker2-API/src/main/java/crafttweaker/api/block/IBlockDefinition.java

```diff
@@ -49,14 +49,22 @@
     void setTickRandomly(boolean tickRandomly);
     
     @ZenMethod
-    void setHarvestLevel(String toolClass, int level);
+    void setHarvestLevel(String toolClass, int level, @Optional IBlockState state);
```

Comment:
1.12 is currently in a state where we accept changes but try to retain bytecode compatibility, as it may be possible (though in this case highly unlikely) that another mod implements this interface as well, or needs to call this method.

Please create a 2nd, default method with all the parameters, that logs an error to the CrT log and keep the old method signature.
You can mark the old one as `@Deprecated` if you like (or instead not use the Optional parameter but overload the method name, as you like)

So, something like
```java
@Deprecated
void setHarvestLevel(String toolClass, int level);

@ZenMethod
default void setHarvestLevel(String toolClass, int level, @Optional IBlockState state) {
    CraftTweakerAPI.logError("Class " + this.getClass().getCanonicalName() + " doesn't override IBlockDefinition::setHarvestLevel!");
}
```

or 
```java
@ZenMethod
void setHarvestLevel(String toolClass, int level);

@ZenMethod
default void setHarvestLevel(String toolClass, int level, IBlockState state) {
    CraftTweakerAPI.logError("Class " + this.getClass().getCanonicalName() + " doesn't override IBlockDefinition::setHarvestLevel!");
}
```